ls8/cpu.c

```diff
@@ -113,18 +113,29 @@ void cpu_run(struct cpu *cpu)
 
       case MUL:
         /* printf("Trying to multiply\n"); */
-        /* cpu->pc += */ 
-        arg_1 = cpu->registers[cpu->pc + 1];
 
-        arg_2 = cpu->registers[cpu->pc + 2];
+        arg_1 = cpu->ram[cpu->pc + 1];
+        /* printf("arg 1: %d\n", arg_1); */
+        arg_2 = cpu->ram[cpu->pc + 2];
+        arg_3 = cpu->ram[cpu->pc + 1];
+        cpu->registers[arg_3] = cpu->registers[arg_1] * cpu->registers[arg_2];
```

Comment:
since `arg_1 == arg_3`, this line could just be
` cpu->registers[arg_1] =* cpu->registers[arg_2];`